### PR TITLE
Include a partition diagram and update verbiage on FPGA MGR interaction

### PR DESCRIPTION
**Pull request by sonals**

### CODE REVIEW

File: src/Documentation/fpga/xrt.rst [hunk 1]
```diff
@@ -471,6 +471,27 @@ these subsystems will be handled in leaf drivers with root and partition drivers
 being part of the infrastructure and provide common services for all leaves found
 on all platforms.
 
+The driver object model looks like the following::
```
Comment (maxzhen) [inline src/Documentation/fpga/xrt.rst:474]: you can remove one extra ':'

File: src/Documentation/fpga/xrt.rst [hunk 2]
```diff
@@ -471,6 +471,27 @@ these subsystems will be handled in leaf drivers with root and partition drivers
...
+
+                    +-----------+
+                    |   root    |
```
Comment (maxzhen) [inline src/Documentation/fpga/xrt.rst:477]: maybe we can put source file name here? root (xmgmt-root.c)?

File: src/Documentation/fpga/xrt.rst [hunk 3]
```diff
@@ -471,6 +471,27 @@ these subsystems will be handled in leaf drivers with root and partition drivers
...
+                    +-----+-----+
+                          |
+              +-----------+-----------+
+              |                       |
+              v                       v
+        +-----------+          +-----------+
+        | partition |    ...   | partition |
```
Comment (maxzhen) [inline src/Documentation/fpga/xrt.rst:484]: also source file name?

File: src/Documentation/fpga/xrt.rst [hunk 4]
```diff
@@ -471,6 +471,27 @@ these subsystems will be handled in leaf drivers with root and partition drivers
...
+        +-----+-----+          +------+----+
+              |                       |
+              |                       |
+        +-----+----+            +-----+----+
+        |          |            |          |
+        v          v            v          v
+    +------+   +------+     +------+   +------+
+    | leaf |...| leaf |     | leaf |...| leaf |
```
Comment (maxzhen) [inline src/Documentation/fpga/xrt.rst:492]: there are many kinds of leaves, so just leave it as "leaf"?

**Review by maxzhen — CHANGES_REQUESTED** [review]
L515 conflicts w/ L566. I think you should remove L515. L566 is correct.

File: src/Documentation/fpga/xrt.rst (hunk 1 above)
Comment (sonals) [reply src/Documentation/fpga/xrt.rst:474]: The :: makes RST treat it as monospace/code and draw a box around it.

File: src/Documentation/fpga/xrt.rst (hunk 2 above)
Comment (sonals) [reply src/Documentation/fpga/xrt.rst:477]: Or maybe put a _info_ box on where to look for code for these objects?

Comment (maxzhen) [reply src/Documentation/fpga/xrt.rst:477]: not in this diagram, but in each related text section?

Comment (sonals) [reply src/Documentation/fpga/xrt.rst:477]: I meant something like a legend which can be done with the note directive
.. note:: 

https://sublime-and-sphinx-guide.readthedocs.io/en/latest/notes_warnings.html

We already use it in our xrt.rst

**Pull request merged**